pytest_catchlog.py

```diff
@@ -162,59 +190,96 @@ def record_tuples(self):
 
             (logger_name, log_level, message)
         """
-        return [(r.name, r.levelno, r.getMessage()) for r in self.records()]
-
-    def set_level(self, level, logger=None):
-        """Sets the level for capturing of logs.
-
-        By default, the level is set on the handler used to capture
-        logs. Specify a logger name to instead set the level of any
-        logger.
-        """
+        return [(r.name, r.levelno, r.getMessage()) for r in self.buffer]
 
-        obj = logger and logging.getLogger(logger) or self.handler
-        obj.setLevel(level)
+    @property
+    def buffer(self):
+        return self._handler.buffer
 
-    def at_level(self, level, logger=None):
-        """Context manager that sets the level for capturing of logs.
+    @property
+    def lines(self):
+        return [r.getMessage() for r in self.buffer]
 
-        By default, the level is set on the handler used to capture
-        logs. Specify a logger name to instead set the level of any
-        logger.
+    def __init__(self, handler):
+        # Do not call LineMatcher.__init__(),
+        # it attempts to assign to self.lines, which is a property.
```

Comment:
My thought on reusing `LineMatcher`:

Pros.
 - It is (widely?) used by plugins for self-testing (catchlog itself, for example); the API is unlikely to change
 - Even if the API changes, that won't affect the main functionality provided by catchlog
 - The logic provided by `LineMatcher` is somewhat hardly related to catchlog to copy-paste the code
 - Reusing code is generally a better option

Cons.
 - The code of `LineMatcher` is a bit dirty, IMO; the API could be more friendly for subclasses
 - It _is_ a dependency, yet not explicit

My opinion: it would be cool to have line matching feature in the core capturing module of pytest (not pytester), i.e. to be able to make assertions about captured stdout/stderr as well.

Let's throw away this feature for now (at least from this PR), and try to think about that in more wide context: (maybe?) move `LineMatcher` from `pytester` to the `capture` plugin, realize what we want to have there, provide backward compatibility, etc.

Anyway, this again is a pytest issue / feature request. But once we have that feature for `capsys`, there would be no doubts to provide the same API for `caplog`.